### DISCUSSION TITLE
Resize by formula

### DISCUSSION THREAD

**Discussion by SoupyGit** [ImageMagick/ImageMagick#5351] [Digital Image Processing] [answered]
Hi I would like to resize an image (in px) according to the following:

-resize ( = Width - Increment )

Where: 
Width = Image Width (from metadata)
Increment = (Start Width - End Width) * Percent
Percent = User defined. Eg. 0.1
End Width = User defined. Eg. 1600
Start Width = User defined. Eg 6000

Math Example:
If Width = 5560
-resize (=5560 - ((6000-1600)*0.1))
-resize (=5560 - 440)
-resize 5120

Is it possible to write a shell script to achieve this?

**Comment by snibgo** [comment, marked as answer]
What version of IM, on what platform? I assume IM 7, usng "magick". The following won't work on IM v6. I assume Windows.

Using the formula you give:
```
-resize %[fx:w-(%StartWidth%-%EndWidth%)*%Percent%]
```
However, "percent" normally means "per hundred", so "0.1 percent" means "one part in one thousand". Your formula suggests it means "proportion".

For Windows BAT, double the first %, hence `%%[fx:...]`.

**Comment by SoupyGit** [comment]
Just for completeness in case anyone else reads, this was correct. The formula for linux is:
-resize "%[fx:w-(${startwidth}-${endwidth})*percent]"
